packages/date-picker/src/DatePicker/components/Calendar/Calendar.tsx

```diff
@@ -106,23 +101,43 @@ export const Calendar: React.VFC<CalendarProps> = ({
         role="dialog"
         aria-modal="true"
       >
-        <DayPicker
-          selectedDays={range ? selectedRange : value}
-          initialMonth={getInitialMonth()}
-          firstDayOfWeek={firstDayOfWeek}
-          disabledDays={disabledDays}
-          onDayClick={onDayChange}
-          navbarElement={getNavbar}
-          className={range ? calendarStyles.range : ""}
-          classNames={defaultCalendarClasses}
-          modifiers={
-            {
-              [calendarStyles.from]: modifiers?.from,
-              [calendarStyles.to]: modifiers?.to,
-            } as Modifiers
-          }
-          onKeyDown={onKeyDown && onKeyDown}
-        />
+        {mode === "single" && (
+          <DayPicker
+            mode="single"
+            selected={value}
+            defaultMonth={getdefaultMonth()}
+            weekStartsOn={weekStartsOn}
+            disabled={disabledDays}
+            onDayClick={onDayChange}
+            classNames={defaultCalendarClasses}
+            components={{
+              IconRight,
+              IconLeft,
+            }}
+          />
+        )}
+        {mode === "range" && (
```

Comment:
With restricting the possibilities of `mode`, we can remove the duplication again.